### PR TITLE
Ignore vendor prefixes reports when a property is not supported

### PR DESCRIPTION
**Pull request by borgitas21**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

## Short description of the change(s)

- [x] #1598
- [x] #1676

### CODE REVIEW

File: packages/hint-compat-api/src/core/api-hint.ts [hunk 1]
```diff
@@ -25,29 +28,29 @@ export abstract class APIHint<T extends Events, K extends Event> implements IHin
         const mdnBrowsersCollection = userBrowsers.convert(context.targetedBrowsers);
 
         this.compatApi = new CompatAPI(namespaceName, mdnBrowsersCollection, isCheckingNotBroadlySupported);
-        this.compatLibrary = new classesMapping[namespaceName](context, this.compatApi.compatDataApi, this.testFeatureIsSupported.bind(this));
+        this.compatLibrary = new classesMapping[namespaceName](context, this.compatApi.compatDataApi, this.testFeature.bind(this));
+
+        (context as HintContext<Events>).on('traverse::end', this.consumeReports.bind(this));
```
[the remark this answers is not in the document]
Comment (borgitas21) [reply packages/hint-compat-api/src/core/api-hint.ts]: I forgot to clean the list of pending reports after being consumed. Now, it should be ok.

**Comment by antross** [comment]
@webhintio/core Per my most recent comments about how this behaves differently depending on the order of the properties: what's everyone's opinion on fixing that now or taking it as a separate bug fix after this PR?

**Comment by antross** [comment]
> we should discuss if we should include #1620 into that change

Agreed.

@borgitas21 What's your opinion on tackling #1620 as an addition to `hint-compat-api`? Do you think it'd benefit from any of the `compat-api` infrastructure or would it be just as easy to do as a standalone hint that depends on `parser-css`?

**Comment by molant** [comment]
> Do you think it'd benefit from any of the compat-api infrastructure

We won't be iterating multiple times in different hints so (slightly?) faster times?

**Comment by borgitas21** [comment]
> Do you think it'd benefit from any of the compat-api infrastructure or would it be just as easy to do as a standalone hint that depends on parser-css?

In terms of performance, I guess it makes sense to use the compat-api so we can prevent to traverse the whole CSS file multiple times. But it can be weird to report malformed CSS from the compat-api hint. So, from my point of view we are mixing things.

**Comment by borgitas21** [comment]
```CSS
.example {
    appearance: none;
    -webkit-appearance: none;
    -moz-appearance: none;
}
```

@antross, I'm not sure, if we must manage scenarios like this one. It's a malformed CSS, and we cannot ensure this is working as expected since the fallback chain is not properly defined. So, I would handle this issue in a separate (or the same Hint), but the compat-api algorithm must assume everything is fine. 

```CSS
.example1 {
    -webkit-appearance: none;
    -moz-appearance: none;
}

.example2 {
    appearance: none;
}
```
However, about the scenario above, it sounds like a corner case, but this is something we must fix since the CSS is "correct".

**Comment by molant** [comment]
> It's a malformed CSS, and we cannot ensure this is working as expected since the fallback chain is not properly defined

It's actually valid css that depending on the browser could generate different results. We would be checking that the order is the one that the order and results are the most compatible ones 🙄

**Comment by antross** [comment]
> In terms of performance, I guess it makes sense to use the compat-api so we can prevent to traverse the whole CSS file multiple times. But it can be weird to report malformed CSS from the compat-api hint. So, from my point of view we are mixing responsibilities.

Agreed. @molant I vote we handle validating the order of prefixed properties as a separate hint. I don't think avoiding the double-walk is going to be much of a performance win and `parser-css` is still only going to parse the file once. That's why we have parsers anyway, no?

That said, I definitely still care about this hint consistently ignoring the prefixes regardless of order - that will help any other validation we do stand out more clearly as we won't be generating extra noise when someone gets it "wrong". Filed as #1676 so we can still wrap up this PR first if needed.

**Comment by antross** [comment]
@webhintio/core Anyone else have time to take a look at this today? I'd like to get it merged as we have a couple of other pieces of work in-progress that are based on it.

**Comment by molant** [comment]
I'll try to take a look today between meetings. @sarvaje maybe you have some cycles today?

**Comment by molant** [comment]
@antross everyone has given the 👍
I'll let you squash and put an appropriate message.

**Pull request merged**